src/ai/storage.py

```diff
@@ -128,6 +128,15 @@ async def store_drone(message: discord.Message, message_copy=None):
         await message.channel.send('You cannot store the Hive Mxtress, silly drone.')
         return False
 
+    # find initiator
+    initiator = fetch_drone_with_drone_id(drone_id)
+
+    # validate specified initiator is message sender
+    if not (drone_id == '0006' and roles.has_role(message.author, roles.HIVE_MXTRESS)):  # temp fix while we decide what to do with the db missing a drone entry for the Hive Mxtress
+        if message.author.id != initiator.id:
+            await message.channel.send(f'You are not {drone_id}. Yes, we can indeed tell identical faceless drones apart from each other.')
+            return False
```

Comment:
This can be reduced to
```suggestion
    # temp fix while we decide what to do with the db missing a drone entry for the Hive Mxtress
    if not (drone_id == '0006' and roles.has_role(message.author, roles.HIVE_MXTRESS)) and message.author.id != initiator.id:
        await message.channel.send(f'You are not {drone_id}. Yes, we can indeed tell identical faceless drones apart from each other.')
        return False
```